* grobian@gentoo.org: we generate bundles, so call them bundles and make
                      sure the linker produces one

--- configure.ac
+++ configure.ac
@@ -184,14 +184,20 @@
   *-*-darwin*)                 # darwin
     shrext=.dylib
     dynamiclib=-dynamiclib
+    bndlext=.bundle
+    bundlelib=-bundle
     ;;
   *-*-mingw* | *-*-cygwin*)    # windows
     shrext=.dll
     dynamiclib=-shared
+    bndlext=.dll
+    bundlelib=-shared
     ;;
   *)                           # linux (and BSD?)
     dynamiclib=-shared
     shrext=.so
+    bundlelib=-shared
+    bndlext=.so
     ;;
 esac
 
@@ -203,6 +209,14 @@
 AC_SUBST(DYNAMICLIB)
 AC_DEFINE_UNQUOTED(DYNAMICLIB, "$dynamiclib", [Dynamic shared library])
 
+BNDLEXT=$bndlext
+AC_SUBST(BNDLEXT)
+AC_DEFINE_UNQUOTED(BNDLEXT,     "$bndlext",     [File extension for shared libraries loaded via dlopen])
+
+BUNDLELIB=$bundlelib
+AC_SUBST(BUNDLELIB)
+AC_DEFINE_UNQUOTED(BUNDLELIB, "$bundlelib", [Dynamic shared library loaded via dlopen])
+
 dnl ===========================================================================
 
 #################
--- extensions/Makefile.am
+++ extensions/Makefile.am
@@ -7,8 +7,8 @@
 endif
 
 CFILES          = $(wildcard $(srcdir)/*.c)
-SOBJS           = $(subst $(srcdir)/,,$(CFILES:.c=$(SHREXT)))
-INSTALLED_ITEMS = $(subst $(srcdir),$(ext_dir),$(CFILES:.c=$(SHREXT)))
+SOBJS           = $(subst $(srcdir)/,,$(CFILES:.c=$(BNDLEXT)))
+INSTALLED_ITEMS = $(subst $(srcdir),$(ext_dir),$(CFILES:.c=$(BNDLEXT)))
 EXTRA_DIST      = $(wildcard *.[ch])
 
 all-local: $(SOBJS)
@@ -20,7 +20,7 @@
 	-I $(top_srcdir)/extensions	\
 	-fPIC
 
-LDFLAGS += $(DYNAMICLIB)
+LDFLAGS += $(BUNDLELIB)
 
 if OS_WIN32
 LDADD    = $(no-undefined) $(libbabldlla) $(MATH_LIB)
@@ -28,20 +28,20 @@
 LDADD    = $(top_builddir)/babl/.libs/libbabl-$(BABL_API_VERSION)$(SHREXT) $(MATH_LIB)
 endif
 
-%$(SHREXT): %.c
+%$(BNDLEXT): %.c
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< $(LDADD)
 
 # if extension needing external libraries are to be # compiled with this make
 # file, each of them can be added according to the this pattern:
-# extra$(SHREXT): extra.c
+# extra$(BNDLEXT): extra.c
 #	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $<  [own compile and link flags]
 
-CIE-Lab$(SHREXT): CIE-Lab.c
+CIE-Lab$(BNDLEXT): CIE-Lab.c
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< $(LDADD)
 
-sse-fixups$(SHREXT): sse-fixups.c
+sse-fixups$(BNDLEXT): sse-fixups.c
 	$(CC) $(CFLAGS) $(MMX_EXTRA_CFLAGS) $(SSE_EXTRA_CFLAGS) $(LDFLAGS) -o $@ $< $(LDADD)
-#lcms$(SHREXT): lcms.c
+#lcms$(BNDLEXT): lcms.c
 #	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< $(LDADD) `pkg-config lcms --cflags --libs`
 #############################################################################
 
--- babl/babl-extension.c
+++ babl/babl-extension.c
@@ -238,7 +238,7 @@
               stat (path, &st);
 
               if ((extension = strrchr (dentry->d_name, '.')) != NULL &&
-                  !strcmp (extension, SHREXT))
+                  !strcmp (extension, BNDLEXT))
                 {
                   babl_extension_load (path);
                 }
